src/cdk-experimental/selection/selection-toggle.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive, Input, OnInit, OnDestroy, ElementRef, ChangeDetectorRef} from '@angular/core';
+import {CdkSelection} from './selection';
+import {Subject} from 'rxjs';
+import {takeUntil} from 'rxjs/operators';
+
+
+@Directive({
+  selector: '[cdkSelectionToggle]',
+  host: {
+    'class': 'cdk-selection-toggle',
+    'role': 'option',
+    'tabindex': '-1',
+    '[attr.aria-selected]': 'selected',
+    '[class.cdk-selection-selected]': 'selected',
+    '(mousedown)': '_onMouseDown($event)',
+    '(click)': '_onToggle()',
+    '(contextmenu)': '_onToggle()',
+    '(keydown.enter)': '_onToggle()',
+    '(keydown.shift)': '_onShift()',
+    '(blur)': '_onBlur()',
+  }
+})
+export class CdkSelectionToggle<T> implements OnInit, OnDestroy {
+
+  /** Model of the item to toggle selection. */
+  @Input('cdkSelectionToggle') model: T | T[];
+
+  /** Whether the selection is disabled or not. */
+  @Input() disabled: boolean;
+
+  /** Whether the toggle is selected or not. */
+  selected: boolean;
+
+  /** The modifier that was invoked. */
+  modifier: 'shift' | 'meta' | null;
+
+  private _destroy = new Subject();
+
+  constructor(
+    public elementRef: ElementRef,
```

Comment:
Does the `elementRef` have to be public?

Reply:
Yes, its used in the parent directive to identify the element order.